src/DssSpell.t.base.sol

```diff
@@ -2834,27 +2834,15 @@ contract DssSpellTestBase is Config, DssTest {
         }
 
         uint256 metadataLength = _getBytecodeMetadataLength(expectedAction);
-        assertTrue(metadataLength <= expectedBytecodeSize, "TestError/metadata-length-gt-expected-bytecode-size");
+        assertLe(metadataLength, expectedBytecodeSize, "TestError/metadata-length-gt-expected-bytecode-size");
         expectedBytecodeSize -= metadataLength;
 
         metadataLength = _getBytecodeMetadataLength(actualAction);
-        assertTrue(metadataLength <= actualBytecodeSize, "TestError/metadata-length-gt-actual-bytecode-size");
+        assertLe(metadataLength, actualBytecodeSize, "TestError/metadata-length-gt-actual-bytecode-size");
         actualBytecodeSize -= metadataLength;
 
         assertEq(actualBytecodeSize, expectedBytecodeSize, "TestError/bytecode-size-mismatch");
-        uint256 size = actualBytecodeSize;
-        uint256 expectedHash;
-        uint256 actualHash;
-        assembly {
-            let ptr := mload(0x40)
-
-            extcodecopy(expectedAction, ptr, 0, size)
-            expectedHash := keccak256(ptr, size)
-
-            extcodecopy(actualAction, ptr, 0, size)
-            actualHash := keccak256(ptr, size)
-        }
-        assertEq(actualHash, expectedHash, "TestError/bytecode-hash-mismatch");
```

Comment:
First of all, I understand why you replaced `assertTrue` with `assertLe` – to make error traces more readable. Fine with me, although it's not directly related to the subject of this PR. But I'm not sure why this code was refactored.

Besides this, looking into the actual change, the code is not the same. Previously, the test only checked hash of the part of the code that is potentially smaller than full code: see how `size` is calculated `uint256 size = extcodesize(actualAction) - _getBytecodeMetadataLength(actualAction)` (`_getBytecodeMetadataLength` currently returns 53). But now we're comparing hashes of the full contract code including metadata (which technically should be a stricter check, unless metadata can be used to cause hash collision). To follow the Chesterton's Fence principle, can you explain why it was important to remove metadata from the contract before generating the hash and no longer is?

In any case, to speed up this PR, you can simply undo this change and submit it separately.

Reply:
> But I'm not sure why this code was refactored.
The reason was because the function was touched, so I took the opportunity to make small improvements.

> To follow the Chesterton's Fence principle, can you explain why it was important to remove metadata from the contract before generating the hash and no longer is?

That's the trap in this case, because there's no documentation around this function, just a couple of disjointed comments that don't give the full context.

We can try to infer that at some point the `solc` compiler would generate different metadata depending on the environment (ie.: OS, architecture, etc.) and the dev who wrote this wanted to prevent that. Other than that, I can't think of any other reason.

Since, as you mentioned, checking the full bytecode is a stricter check and it also reduces the complexity, I thought it would be a good idea to make this change.

In any case, if we're still willing to follow Chesterton's Fence Principle to the letter, I can revert them to not stall this PR.